### PR TITLE
Feat/unc func pytest

### PR DESCRIPTION
**Pull request by davidegraff**
## Description
reformatted the unit tests to use `pytest`. The platform is a lot more flexible and I want to enforce that all new tests use pytest if possible.

## Checklist
- [x] linted with flake8?
- [x] (if appropriate) unit tests added?

### CODE REVIEW

File: tests/test_unit/test_uncertainty_evaluators.py [hunk 1]
```diff
@@ -1,159 +1,126 @@
...
+    )
+
+
+@pytest.fixture
+def spearman_evaluator():
+    return build_uncertainty_evaluator("spearman", None, "ensemble", "regression", "mse", None)
+
+
+def test_build_regression_metric(regression_metric):
+    assert build_uncertainty_evaluator(regression_metric, None, None, "regression", None, None)
```
Comment (cjmcgill) [inline tests/test_unit/test_uncertainty_evaluators.py:55]: Previously, this test checked to see if the output was an `UncertaintyEvaluator`. Now it's changed to be checking just that it's not None or False right?

I can't think of a corner case where the distinction would matter I suppose. So I think it's okay.

Comment (cjmcgill) [reply tests/test_unit/test_uncertainty_evaluators.py:55]: Also this request structure is interesting. Wasn't familiar with it previously.

File: tests/test_unit/test_uncertainty_evaluators.py [hunk 2]
```diff
@@ -1,159 +1,126 @@
...
+
+
+def test_build_multiclass_metric(multiclass_metric):
+    assert build_uncertainty_evaluator(multiclass_metric, None, None, "multiclass", None, None)
+
+
+@pytest.mark.parametrize("metric", [str(uuid.uuid4()) for _ in range(3)])
+def test_build_unsupported_metrics(metric, dataset_type):
+    with pytest.raises(NotImplementedError):
+        build_uncertainty_evaluator(metric, None, None, "spectra", dataset_type, None)
```
Comment (cjmcgill) [inline tests/test_unit/test_uncertainty_evaluators.py:71]: I don't understand what's going on in this block. Is "metric" just a string or is pytest doing something funny with it where it becomes the name of an attribute or something?

And is the uuid just putting gibberish that then becomes dataset_type (though dataset_type is in the loss_function spot).

This is all just to trigger L398 from the "spectra" entry, so why the gibberish?

**Review by cjmcgill — APPROVED** [review]
Okay looks good
